pkg/apis/messaging/v1beta1/kafka_channel_lifecycle.go

```diff
@@ -99,8 +99,18 @@ func (kcs *KafkaChannelStatus) GetCondition(t apis.ConditionType) *apis.Conditio
 }
 
 // IsReady returns true if the resource is ready overall.
-func (kcs *KafkaChannelStatus) IsReady() bool {
-	return kcs.GetConditionSet().Manage(kcs).IsHappy()
+func (kc *KafkaChannel) IsReady() bool {
+	kcs := kc.Status
+	return kcs.ObservedGeneration == kc.Generation &&
+		kcCondSet.Manage(&kcs).IsHappy()
+}
+
+// IsFailed returns true if the resource has observed the latest generation
+// and ready is false.
+func (kc *KafkaChannel) IsFailed() bool {
+	kcs := kc.Status
+	return kcs.ObservedGeneration == kc.Generation &&
+		!kcCondSet.Manage(&kcs).IsHappy()
```

Comment:
So isFailed() will be false (indicating... it's "not-failed" or "ready"?) if the generations don't match ?   I might be mis-interpreting but it seems confusing?

Reply:
isFailed is TRUE when the isREADY is false,
see: https://github.com/knative-sandbox/eventing-kafka/pull/825/commits/01abb66d419e2573a1d7c9a08a8d22d1486bfeec

Reply:
Right but...  I meant that both `isReady()` AND `isFailed()` can be **false** at the same time (if generations don't match) - which means `!isFailed()` does not imply `isReady() == true`.  Otherwise we could just write `isFailed()` as `return !isReady()`.

Ahmed mentioned this was based on precedent in other Knative implementations?  I'm not saying it's a blocker for the PR necessarily - it is just confusing to try and represent a trifecta of states using two functions, where the naming is short and thus requires understanding the inner implementation.

Reply:
Maybe this is all fine then (per Ahmed's clarifications below)?   Would capturing that as comments on the two functions be helpful?